apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs

```diff
@@ -255,5 +187,78 @@ internal Job CreateJob(JobConfiguration configuration, JobCreationOptions option
+        {
+            GaxPreconditions.CheckNotNull(jobReference, nameof(jobReference));
+            var request = Service.Jobs.Cancel(jobReference.ProjectId, jobReference.JobId);
+            request.ModifyRequest += _versionHeaderAction;
+            options?.ModifyRequest(request);
+            return request;
+        }
+
+        private InsertRequest CreateExtractJobRequest(TableReference tableReference, IEnumerable<string> destinationUris, CreateExtractJobOptions options)
+        {
+            GaxPreconditions.CheckNotNull(tableReference, nameof(tableReference));
+            GaxPreconditions.CheckNotNull(destinationUris, nameof(destinationUris));
+            List<string> destinationUriList = destinationUris.ToList();
+            GaxPreconditions.CheckArgument(destinationUriList.Count != 0, nameof(destinationUris), "Destination URIs cannot be empty");
+
+            var extract = new JobConfigurationExtract { DestinationUris = destinationUriList, SourceTable = tableReference };
+            options?.ModifyRequest(extract);
+            return CreateInsertJobRequest(new JobConfiguration { Extract = extract }, options);
+        }
+
+        private InsertRequest CreateCopyJobRequest(IEnumerable<TableReference> sources, TableReference destination, CreateCopyJobOptions options)
+        {
+            GaxPreconditions.CheckNotNull(sources, nameof(sources));
+            GaxPreconditions.CheckNotNull(destination, nameof(destination));
+            List<TableReference> sourceList = sources.ToList();
+            GaxPreconditions.CheckArgument(sourceList.Count != 0, nameof(sources), "Sources cannot be empty");
+
+            var copy = new JobConfigurationTableCopy { SourceTables = sourceList, DestinationTable = destination };
+            options?.ModifyRequest(copy);
+            return CreateInsertJobRequest(new JobConfiguration { Copy = copy }, options);
+        }
+
+        private InsertRequest CreateLoadJobRequest(IEnumerable<string> sourceUris, TableReference destination, TableSchema schema, CreateLoadJobOptions options)
+        {
+            GaxPreconditions.CheckNotNull(sourceUris, nameof(sourceUris));
+            GaxPreconditions.CheckNotNull(destination, nameof(destination));
+            List<string> sourceList = sourceUris.ToList();
+            GaxPreconditions.CheckArgument(sourceList.Count != 0, nameof(sourceUris), "Source URIs collection cannot be empty");
+
+            var load = new JobConfigurationLoad { SourceUris = sourceList, DestinationTable = destination, Schema = schema };
+            options?.ModifyRequest(load);
+            var request = CreateInsertJobRequest(new JobConfiguration { Load = load }, options);
+            return request;
+        }
+
+        private InsertRequest CreateInsertJobRequest(JobConfiguration configuration, JobCreationOptions options)
+        {
+            var job = CreateJob(configuration, options);
```

Comment:
Check configuration isn't null?

Reply:
CreateJob does that already - and everything that calls this method news up a configuration anyway.